tests/unit/s2n_tls13_parse_record_type_test.c

```diff
@@ -132,22 +132,116 @@ int main(int argc, char **argv)
 
         EXPECT_SUCCESS(s2n_tls13_parse_record_type(&stuffer, &record_type));
         EXPECT_EQUAL(record_type, 0x16);
+        /* There was no data before the record type */
+        EXPECT_EQUAL(s2n_stuffer_data_available(&stuffer), 0);
+
+        EXPECT_SUCCESS(s2n_stuffer_free(&stuffer));
+    }
+
+    /* Test maximum record length size (without padding) */
+    {
+        struct s2n_stuffer stuffer;
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 1024));
+
+        /* fill up stuffer to before the limit */
+        while (s2n_stuffer_data_available(&stuffer) < S2N_MAXIMUM_INNER_PLAINTEXT_LENGTH) {
+            EXPECT_SUCCESS(s2n_stuffer_write_uint8(&stuffer, 0x16));
+        }
+        EXPECT_EQUAL(s2n_stuffer_data_available(&stuffer), S2N_MAXIMUM_INNER_PLAINTEXT_LENGTH);
+
+        EXPECT_SUCCESS(s2n_tls13_parse_record_type(&stuffer, &record_type));
+        EXPECT_EQUAL(record_type, 0x16);
+        /* The last byte is stripped as the content type */
+        EXPECT_EQUAL(s2n_stuffer_data_available(&stuffer), S2N_MAXIMUM_INNER_PLAINTEXT_LENGTH - 1);
+
+        EXPECT_SUCCESS(s2n_stuffer_free(&stuffer));
+    }
+
+    /* Certain versions of Java can generate inner plaintexts with lengths up to
+     * S2N_MAXIMUM_INNER_PLAINTEXT_LENGTH + 16 (See JDK-8221253)
+     * However, after the padding is stripped, the result will always be no more than
+     * S2N_MAXIMUM_INNER_PLAINTEXT_LENGTH - 1
+     */
```

Comment:
To make it clear which tests are dealing with this Java problem, you could consider nesting them in another { } context